gapis/service/memory_box/box.go

```diff
@@ -248,15 +255,41 @@ func Box(ctx context.Context, d *memory.Decoder, t *types.Type) (*Value, error)
 		}
 	case *types.Type_Pseudonym:
 		if elem, ok := types.TryGetType(t.Pseudonym.Underlying); ok {
-			return Box(ctx, d, elem)
+			b, err := Box(ctx, d, elem, p, rc)
+			if err != nil {
+				return nil, err
+			}
+
+			reflType, err := types.GetReflectedType(typeId)
+			if err != nil {
+				return nil, err
+			}
+
+			if reflType.Implements(tyLinkable) {
+				// reflectively convert the underlying value back to what its
+				// api type should have been, then attempt to link through it.
+				v := reflect.New(reflType)
+				switch vv := b.Val.(*Value_Pod).Pod.Val.(type) {
+				case (*pod.Value_Uint64):
+					v.Elem().SetUint(vv.Uint64)
+				case (*pod.Value_Uint32):
+					v.Elem().SetUint(uint64(vv.Uint32))
```

Comment:
we still should fail if in the default case, no?

Reply:
In the default case we'll end up leaving v as its zero value and so we won't produce a link. I'm not sure if blowing up the request is preferable.

Reply:
True, ok, no default.